pkg/webhook/handler.go

```diff
@@ -18,16 +19,50 @@ type Handler interface {
 	admission.Handler
 	admission.DecoderInjector
 	inject.Client
+	// Initialize and validate the handler
+	Initialize() error
+}
+
+// withValidationHandler create a validation handler instance
+func withValidationHandler(validator Validator, object runtime.Object) Handler {
+	return &handler{validator: validator, Object: object}
```

Comment:
I would recommend to just set `injector` aka `handler` here, I think public accessible internalizer isn't needed as the handler is an unexported type anyway.